h2o-algos/src/main/java/hex/tree/drf/DRF.java

```diff
@@ -155,8 +163,23 @@ private void initTreeMeasurements() {
       growTrees(ktrees, leafs, _rand);
 
       // Move rows into the final leaf rows - fill "Tree" and OUT_BAG_TREES columns and zap the NIDs column
-      CollectPreds cp = new CollectPreds(ktrees,leafs,_model.defaultThreshold()).doAll(_train,_parms._build_tree_one_node);
+      double[] idx = new double[(int)_train.numRows()];
+      for(int i = 0; i < _train.numRows(); i++) {
+        idx[i] = i;
+      }
+      
+      boolean collectConfidenceData = true; // TODO parameter
+
+      String id_for_confidence_metrics = "id_for_confidence_metrics";
+      if(collectConfidenceData && _train.find(id_for_confidence_metrics) == -1) {
+        _train.add(id_for_confidence_metrics, Vec.makeCon(Vec.newKey(), idx));
+      }
+      int indexOfIdColumn = _train.find(id_for_confidence_metrics);
+      CollectPreds cp = new CollectPreds(ktrees,leafs,_model.defaultThreshold(), indexOfIdColumn).doAll(_train,_parms._build_tree_one_node);
 
+      IcedHashMap<Integer, Double[]> predictionsMap = cp._predictionsMap;
```

Comment:
Why do we need Double[] array? Can we use a primitive array instead?

Reply:
@michalkurka At the beginning I was using different types for different parameters and my Map was storing values as Object[]. Later when I added Surrogate Model I needed to convert my grid  into Frame and then be able to extract rows from this Frame....and there are only `.atd(index)`, `.at8(index)` or something like this which return only doubles or long. So I decided to use single representation with doubles. Yes with current version probably there is no point in having Doubles.